libreyolo/models/sam/base.py

```
@@ -0,0 +1,604 @@
+        """Always fp32 — including on CUDA.
+
+        Unlike the (multi-GB) VLM tier, SAM-1 checkpoints are small enough that
+        fp32 inference is cheap, and half precision actively hurts SAM: mask
+        logits lose boundary fidelity, and — more importantly — bf16's 8-bit
+        mantissa rounds prompt-point coordinates by several pixels once they
+        exceed 256 (the model frame goes up to 1024), silently shifting where
+        the user clicked. Half precision can return later as an explicit opt-in.
+        """
+        return torch.float32
+
+    def _load_pretrained(self, snapshot_dir: str):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_load_pretrained`

(ANN202)